API/ev3_sensors/c4ev3_compatibility.c

```diff
@@ -68,11 +69,24 @@ int ReadSensor(int port) {
         case TOUCH_PRESS:
             return ReadEV3TouchSensor(port);
         case COL_REFLECT:
-            return ReadEV3ColorSensorLight(port, ReflectedLight);
+            return ReadEV3ColorSensorReflectedLight(port);
         case COL_AMBIENT:
-            return ReadEV3ColorSensorLight(port, AmbientLight);
+            return ReadEV3ColorSensorAmbientLight(port);
         case COL_COLOR:
-            return ReadEV3ColorSensor(port);
+            return ReadEV3ColorSensorColor(port);
+        case COL_COLOR_RGB: {
+#define TO_8BIT(x) (((x) >> 2) & 0xFF)
+
+            RGB colors = {0};
+            int retval = ReadEV3ColorSensorColorRGB(port, &colors);
+
+            if (retval < 0)
+                return INT_MIN;
+
+            return (TO_8BIT(colors.red) << 16)
+                 | (TO_8BIT(colors.green) << 8)
+                 | (TO_8BIT(colors.blue));
+        }
```

Comment:
Do you think it could be useful to also provide three macros to get the RGB components?

I was thinking about something like this (maybe in the c4ev3_compatibility.h file):
```c
#define RGB_INT_GET_RED(x) (((x) >> 16) & 0xFF)
#define RGB_INT_GET_GREEN(x) (((x) >> 8) & 0xFF)
#define RGB_INT_GET_BLUE(x) ((x) & 0xFF)
```

Reply:
Yup, I think too it would be useful. I will add it to that header.